traffic_stats/influxdb_tools/sync/sync_test.go

```diff
@@ -0,0 +1,230 @@
+/*
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+*/
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"testing"
+
+	influx "github.com/influxdata/influxdb/client/v2"
+	"github.com/influxdata/influxdb/models"
+	. "github.com/smartystreets/goconvey/convey"
```

Comment:
I really do think that in general, having something that can handle assertions cleanly in some way is generally a lot mare palatable to read and write than just the straight `testing` package. Another alternative to goconvey could be [testify](https://github.com/stretchr/testify). I haven't used it much but it has some of the same advantages from goconvey, namely that instead of having something like
```go
if err != nil {
   t.Fail()
}
```
it becomes 
``` go
assert.NotNil(t, err)
```

This is even more nicer when operating on structured objects, because you can do deep comparisons, something like

```go
expectedResp := &influx.Response{ ... }
actualResp := influxClient.Query(...)
assert.Equal(t, actualResp, expectedResp)
```
which obviously cuts down the number of lines proportional to the number of fields you'd otherwise have to check manually.

Of course, we could go with neither framework, but my two cents in general are that making tests easy to add typically encourages people to write their own.